src/components/study/programming-language-section.test.tsx

```diff
@@ -0,0 +1,328 @@
+import { describe, it, expect, vi, beforeEach, Mock } from 'vitest'
+import { renderWithProviders, screen, faker, userEvent } from '@/tests/unit.helpers'
+import { useQuery } from '@/common'
+import { useSession } from '@/hooks/session'
+import { UseFormReturnType } from '@mantine/form'
+import { StudyProposalFormValues } from '@/app/[orgSlug]/study/request/study-proposal-form-schema'
+import { ProgrammingLanguageSection } from './programming-language-section'
+
+vi.mock('@/common', async () => {
+    const actual = await vi.importActual('@/common')
+    return {
+        ...actual,
+        useQuery: vi.fn(),
+    }
+})
+
+vi.mock('@/hooks/session', () => ({
+    // By default tests will configure this to return a non-admin session. Individual tests
+    // can override the return value to simulate admin reviewers for the selected org.
+    useSession: vi.fn(),
+}))
+
+// Helper to create a mock org with supported languages
+const createMockOrg = (
+    overrides: Partial<{
+        slug: string
+        name: string
+        type: string
+        supportedLanguages: Array<'R' | 'PYTHON'>
+    }> = {},
+) => ({
+    slug: overrides.slug ?? faker.string.alpha(10),
+    name: overrides.name ?? 'Test Org',
+    type: overrides.type ?? 'enclave',
+    supportedLanguages: overrides.supportedLanguages ?? [],
+})
+
+// Helper to create a mock form
+const createMockForm = (
+    overrides: Partial<{
+        orgSlug: string
+        language: 'R' | 'PYTHON' | null
+    }> = {},
+): UseFormReturnType<StudyProposalFormValues> => {
+    const values = {
+        orgSlug: overrides.orgSlug ?? '',
+        language: overrides.language ?? null,
+        title: 'Test Study',
+        piName: 'Test PI',
+        descriptionDocument: null,
+        irbDocument: null,
+        agreementDocument: null,
+        mainCodeFile: null,
+        additionalCodeFiles: [],
+    }
+
+    const form = {
+        values,
+        errors: {},
+        setFieldValue: vi.fn(),
+        getInputProps: vi.fn(),
+        // Add minimal form methods needed
+    } as unknown as UseFormReturnType<StudyProposalFormValues>
+
+    return form
+}
+
```

Comment:
this is mocking so much that I question what value the test is giving us?  Does the e2e study flow test cover this? If so I think we can  do not need this complexity